vertical-pod-autoscaler/pkg/utils/limitrange/limit_range_calculator.go

```diff
@@ -55,12 +56,12 @@ func NewLimitsRangeCalculator(f informers.SharedInformerFactory) (*limitsChecker
 	}
 	limitRangeLister := f.Core().V1().LimitRanges().Lister()
 	stopCh := make(chan struct{})
-	f.Start(stopCh)
-	for _, ok := range f.WaitForCacheSync(stopCh) {
-		if !ok {
-			if !f.Core().V1().LimitRanges().Informer().HasSynced() {
-				return nil, fmt.Errorf("informer did not sync")
-			}
+	informer := f.Core().V1().LimitRanges().Informer()
+	go informer.Run(stopCh)
+	ok := cache.WaitForCacheSync(stopCh, informer.HasSynced)
+	if !ok {
+		if !f.Core().V1().LimitRanges().Informer().HasSynced() {
```

Comment:
do we really need this inner `if`? It seems to be redundant with `cache.WaitForCacheSync(stopCh, informer.HasSynced)`